### PR TITLE
arm: AArch64: Save the callee-saved register directly into SP

### PR DESCRIPTION
**Pull request by carlocaione**
This PR is taken out from an old discussion at #22877. I still think this is a change worth to have especially because soon I'll be start working again on the switch to the new arch_switch().

### CODE REVIEW

**Comment by carlocaione** [comment]
tagging a few people: @stephanosio @Abhishek-brcm @wentongwu @sandeepbrcm

**Comment by carlocaione** [comment]
Converting this to a draft for now because I'm reworking a bit the assembly code and this will probably be unnecessary.

File: arch/arm/include/aarch64/stack.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+struct _callee_saved_stack {
+	uint64_t x29, x30;
+	uint64_t x27, x28;
+	uint64_t x25, x26;
+	uint64_t x23, x24;
+	uint64_t x21, x22;
+	uint64_t x19, x20;
+};
+
+typedef struct _callee_saved_stack _callee_saved_stack_t;
```
Comment (andrewboie) [inline arch/arm/include/aarch64/stack.h:36]: why the use of typedef here? don't see this being used in an offsets.c. just use the struct directly.

**Review by wentongwu — COMMENTED** [review]
I'm not sure if there is security issue saving callee-saved registers directly to stack.

**Comment by carlocaione** [comment]
This change is not needed anymore. Closing for now.

**Pull request closed without merging**